sphinx_immaterial/__init__.py

```diff
@@ -269,6 +280,15 @@ def setup(app):
         "html_use_directory_uris_for_index_pages", False, rebuild="html", types=bool
     )
 
+    app.add_node(
```

Comment:
Perhaps make `dropdown.py` usable as a sphinx extension on its own, by defining a `setup` function there?

Reply:
Its a workaround. Ideally, we could use other sphinx extensions (much better designed and maintained elsewhere) if we remove the CSS rule for `details` and `summary` tags.

Reply:
I have defined the setup function in dropdown.py. Now, \_\_init\_\_.py imports the setup function (under an alias) and executes it within the `__init__.setup()`

I'm not sure how much use the new directive would be as an independent extension because it largely is based on the CSS inherited from mkdocs-material.

Reply:
My thinking was not so much that it would be used independently, but rather that it would be an optional feature.  This way it is fully self-contained.

Reply:
Hmm, that would also incur optionally `@import`ing the CSS about `details` & `summary`. That sounds nice, but I think it would more work to make it actually self-contained. 

If we remove CSS about `details`/`summary`, then the dropdown directive would be contained in a completely unrelated extension like sphinx-design. I'm all for that because we already have to rely on them for `set-tab` blocks (& hopefully inline icons in the future).

Reply:
Sounds like the best option is to actually just remove the details/summary CSS from this theme then.